crates/turborepo-lib/src/run/summary/task_factory.rs

```diff
@@ -187,7 +205,10 @@ impl<'a> TaskSummaryFactory<'a> {
                 self.env_at_start,
             )
             .expect("invalid glob in task definition should have been caught earlier"),
-            dot_env: task_definition.dot_env.clone(),
+            dot_env: match task_definition.dot_env.is_empty() {
```

Comment:
I feel like this won't work because when we process dotEnv, we flatten both the `None` case and the empty array case into an empty vector. So if you put `dotEnv: []` or if you put `dotEnv: null` (or omit the field entirely) in your turbo.json, we'll output `null` in both cases here. But in Go, we'll output `null` when we pass `dotEnv: null` and `[]` when we pass `dotEnv: []`.

Reply:
Good catch